Foundation/src/Environment_WIN32U.cpp

```diff
@@ -209,31 +209,65 @@ void EnvironmentImpl::nodeIdImpl(NodeId& id)
 	// Make an initial call to GetAdaptersInfo to get
 	// the necessary size into len
 	DWORD rc = GetAdaptersInfo(pAdapterInfo, &len);
-	if (rc == ERROR_BUFFER_OVERFLOW)
-	{
-		delete [] reinterpret_cast<char*>(pAdapterInfo);
-		pAdapterInfo = reinterpret_cast<IP_ADAPTER_INFO*>(new char[len]);
-	}
-	else if (rc != ERROR_SUCCESS)
-	{
-		delete[] reinterpret_cast<char*>(pAdapterInfo);
-		throw SystemException("cannot get network adapter list");
-	}
-	if (GetAdaptersInfo(pAdapterInfo, &len) == NO_ERROR)
+
+	try
 	{
-		pAdapter = pAdapterInfo;
+		if (rc == ERROR_BUFFER_OVERFLOW)
+		{
+			delete[] reinterpret_cast<char*>(pAdapterInfo);
+			pAdapterInfo = reinterpret_cast<IP_ADAPTER_INFO*>(new char[len]);
+		}
+		else if (rc != ERROR_SUCCESS)
+		{
+			throw SystemException("cannot get network adapter list");
+		}
+
 		bool found = false;
-		while (pAdapter && !found)
+
+		if (GetAdaptersInfo(pAdapterInfo, &len) == NO_ERROR)
 		{
-			if (pAdapter->Type == MIB_IF_TYPE_ETHERNET && pAdapter->AddressLength == sizeof(id))
+			pAdapter = pAdapterInfo;
+
+			while (pAdapter && !found)
 			{
-				found = true;
-				std::memcpy(&id, pAdapter->Address, pAdapter->AddressLength);
+				if (pAdapter->Type == MIB_IF_TYPE_ETHERNET && pAdapter->AddressLength == sizeof(id))
+				{
+					found = true;
+					std::memcpy(&id, pAdapter->Address, pAdapter->AddressLength);
+				}
+				pAdapter = pAdapter->Next;
 			}
-			pAdapter = pAdapter->Next;
+
+			// if an ethernet adapter was not found, search for a wifi adapter
+			pAdapter = pAdapterInfo;
+
+			while (pAdapter && !found)
+			{
+				if (pAdapter->Type == IF_TYPE_IEEE80211 && pAdapter->AddressLength == sizeof(id))
+				{
+					found = true;
+					std::memcpy(&id, pAdapter->Address, pAdapter->AddressLength);
+				}
+				pAdapter = pAdapter->Next;
+			}
+		}
+		else
+		{
+			throw SystemException("cannot get network adapter list");
 		}
+
+		// ethernet and wifi adapters not found, fail the search
+		if (!found)
+		{
+			throw SystemException("No ethernet or wifi adapter found");
```

Comment:
Some exception strings start with capital word, some don't. IMO they shall be unified. Majority of strings used with SystemException are lower case.

Reply:
Changed this one and the ones in Environment_UNIX.cpp to lower case.